File: rev-com-server/src/main/java/com/codergb/bean/movie/Screenwriter.java
package com.codergb.bean.movie;

import com.codergb.bean.FileInfo;
import com.codergb.bean.occupation.Occupation;
import lombok.*;

import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@EqualsAndHashCode(callSuper = true)
public class Screenwriter extends FileInfo {
  private String id;
  private String name;
  private String avatarUrl;
  private String alias;
  private String family;
  private List<Occupation> occupations;
  private String gender;
  private String birthPlace;
  private Date birth;
  private String description;
  private Date createTime;
  private Date updateTime;
}

File: rev-com-client/src/types/movie/screenwriter.ts
import { IOccupation } from "./occupation"

interface IScreenwriter {
  id: string
  name: string
  avatarUrl: string
  alias: string
  family: string
  occupations: IOccupation[]
  gender: string
  birthPlace: string
  birth: string
  description: string
  createTime: string
  updateTime: string
}
export type { IScreenwriter }
